xm-commons-function/src/main/java/com/icthh/xm/commons/config/FunctionApiSpecsProcessor.java

```diff
@@ -0,0 +1,41 @@
+package com.icthh.xm.commons.config;
+
+import com.icthh.xm.commons.domain.spec.FunctionApiSpecs;
+import com.icthh.xm.commons.domain.spec.FunctionSpec;
+import com.icthh.xm.commons.processor.impl.DefinitionSpecProcessor;
+import com.icthh.xm.commons.processor.impl.FormSpecProcessor;
+import com.icthh.xm.commons.service.DefaultSpecProcessingService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class FunctionApiSpecsProcessor extends DefaultSpecProcessingService<FunctionApiSpecs> {
+
+    private final DefinitionSpecProcessor definitionSpecProcessor;
+    private final FormSpecProcessor formSpecProcessor;
+
+    public FunctionApiSpecsProcessor(DefinitionSpecProcessor definitionSpecProcessor,
+                                     FormSpecProcessor formSpecProcessor) {
+        super(definitionSpecProcessor, formSpecProcessor);
+        this.definitionSpecProcessor = definitionSpecProcessor;
+        this.formSpecProcessor = formSpecProcessor;
+    }
+
+    @Override
+    public FunctionApiSpecs processSpecification(String tenant, String dataSpecKey, FunctionApiSpecs specification) {
+        specification.getSpecifications().forEach(f -> {
+            processValidateInputParameter(specification.isValidateFunctionInput(), f);
```

Comment:
can ve override just processDataSpecification ?

Reply:
yes, see default method in SpecificationProcessingService:
`default S processSpecification(String tenant, String dataSpecKey, S specification) {
        processDataSpecification(tenant, dataSpecKey, specification.getItems());
        return specification;
    }`